src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -138,14 +149,33 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
   /* The autocomplete panel to be attached to this trigger. */
   @Input('matAutocomplete') autocomplete: MatAutocomplete;
 
+  /** Stream of clicks outside of the autocomplete panel. */
+  private _outsideClickStream: Observable<any>;
+
   constructor(private _element: ElementRef, private _overlay: Overlay,
               private _viewContainerRef: ViewContainerRef,
               private _zone: NgZone,
               private _changeDetectorRef: ChangeDetectorRef,
               @Inject(MAT_AUTOCOMPLETE_SCROLL_STRATEGY) private _scrollStrategy,
               @Optional() private _dir: Directionality,
               @Optional() @Host() private _formField: MatFormField,
-              @Optional() @Inject(DOCUMENT) private _document: any) {}
+              @Optional() @Inject(DOCUMENT) document: any) {
+    this._nextStable = first.call(_zone.onStable);
+    this._outsideClickStream = document ?
```

Comment:
Can you move this _outsideClickStream assignment out into a function?

Reply:
I think it makes more sense for the assignment to be here. Mainly, we can avoid having document be a class property and make this._outsideClickStream a readonly property. This has the added benefit of making sure we only emit once per a click event. Previously, there were two emissions. And it would actually emit something different on the second emission for the same click because the filter body was called twice.

Reply:
For code style, we prefer not to have this much bloat in the constructor.  Could you not keep the assignment here and move the RHS to a normal function call? 

```ts
this._outsideClickStream = this._getOutsideClickStream(document);
```

Reply:
Okay, this accomplishes the same thing. And you're right, this does look much nicer. Change was made.